net/api/test.cc

```diff
@@ -237,13 +278,108 @@ TEST(HTTPAPI, PostFromBufferToBuffer) {
   HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
   HTTP(FLAGS_net_api_test_port).Register("/post", [](Request r) {
     ASSERT_TRUE(r.http.HasBody());
-    r.connection.SendHTTPResponse("Data: " + r.http.Body());
+    r("Data: " + r.http.Body());
   });
   const auto response =
       HTTP(POST(Printf("localhost:%d/post", FLAGS_net_api_test_port), "No shit!", "application/octet-stream"));
   EXPECT_EQ("Data: No shit!", response.body);
 }
 
+TEST(HTTPAPI, PostWithEmptyBody) {
+  HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
+  HTTP(FLAGS_net_api_test_port).Register("/post", [](Request r) {
+    ASSERT_FALSE(r.http.HasBody());
+    r("Empty POST.");
+  });
+  EXPECT_EQ("Empty POST.", HTTP(POST(Printf("localhost:%d/post", FLAGS_net_api_test_port))).body);
+}
+
+TEST(HTTPAPI, PostAStringAsString) {
+  HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
+  HTTP(FLAGS_net_api_test_port).Register("/post_string", [](Request r) {
+    ASSERT_TRUE(r.http.HasBody());
+    r(r.http.Body());
+  });
+  EXPECT_EQ("std::string",
+            HTTP(POST(Printf("localhost:%d/post_string", FLAGS_net_api_test_port),
+                      std::string("std::string"),
+                      "text/plain")).body);
+}
+
+TEST(HTTPAPI, PostAStringAsConstCharPtr) {
+  HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
+  HTTP(FLAGS_net_api_test_port).Register("/post_const_char_ptr", [](Request r) {
+    ASSERT_TRUE(r.http.HasBody());
+    r(r.http.Body());
+  });
+  EXPECT_EQ("const char*",
+            HTTP(POST(Printf("localhost:%d/post_const_char_ptr", FLAGS_net_api_test_port),
+                      static_cast<const char*>("const char*"),
+                      "text/plain")).body);
+}
+
+TEST(HTTPAPI, RespondWithStringAsString) {
+  HTTP(FLAGS_net_api_test_port).ResetAllHandlers();
+  HTTP(FLAGS_net_api_test_port).Register("/respond_with_std_string", [](Request r) {
+    ASSERT_FALSE(r.http.HasBody());
+    r.connection.SendHTTPResponse(std::string("std::string"), HTTPResponseCode::InternalServerError);
```

Comment:
Lack of sleep is a terrible thing, you know? Ah, wait, never mind, you do :( Fixed!